lightningd/chaintopology.c

```diff
@@ -955,9 +957,11 @@ static void topo_add_utxos(struct chain_topology *topo, struct block *b)
 			if (!amount_asset_is_main(&amt))
 				continue;
 
-			const u8 *script = bitcoin_tx_output_get_script(tmpctx, tx, n);
-			if (is_p2wsh(script, NULL)) {
+			if (bitcoin_tx_output_script_is_p2wsh(tx, n)) {
+				outpoint.txid = b->txids[i];
```

Comment:
If you are going to re-copy the txid, you should remove it from the init of `struct bitcoin_outpoint outpoint` above, at which point you might as well move the struct definition here and init it inline.

Reply:
As far as I can see the txid is copied only once, and the declaration at the top of the function reserves room, but doesn't initialize anymore.